crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -107,67 +118,38 @@ impl TransactionBlock {
     /// If the owner of the gas object(s) is not the same as the sender, the transaction block is a
     /// sponsored transaction block.
     async fn gas_input(&self) -> Option<GasInput> {
-        Some(GasInput::from(self.native.transaction_data().gas_data()))
+        Some(GasInput::from(self.native().gas_data()))
     }
 
     /// The type of this transaction as well as the commands and/or parameters comprising the
     /// transaction of this kind.
     async fn kind(&self) -> Option<TransactionBlockKind> {
-        Some(TransactionBlockKind::from(
-            self.native.transaction_data().kind().clone(),
-        ))
+        Some(TransactionBlockKind::from(self.native().kind().clone()))
     }
 
     /// A list of all signatures, Base64-encoded, from senders, and potentially the gas owner if
     /// this is a sponsored transaction.
     async fn signatures(&self) -> Option<Vec<Base64>> {
-        Some(
-            self.native
-                .tx_signatures()
+        self.native_signed_data().map(|s| {
+            s.tx_signatures()
                 .iter()
-                .map(|s| Base64::from(s.as_ref()))
-                .collect(),
-        )
+                .map(|sig| Base64::from(sig.as_ref()))
+                .collect()
+        })
     }
 
     /// The effects field captures the results to the chain of executing this transaction.
     async fn effects(&self) -> Result<Option<TransactionBlockEffects>> {
         Ok(Some(
-            TransactionBlockEffects::try_from(self.stored.clone()).extend()?,
+            TransactionBlockEffects::try_from(self.clone()).extend()?,
```

Comment:
nit: based on how this is written on the `TransactionBlockEffects` side, can this be written as `self.clone().try_into().extend()?`?